### ISSUE TITLE
✨ dagger cli should have `cache` subcommands

### ISSUE THREAD

**Issue by jedevc** [dagger/dagger#6260]
### What are you trying to do?

Discussed in discord with @vito and @marcosnils: https://discord.com/channels/707636530424053791/1003718839739105300/1184142736664838265

It would be nice to have some subcommands for the dagger cli that can help inspect and prune the cache. At a basic level, it would be nice to have similar behavior to `buildctl du` and `buildctl prune`.

Future extension ideas:
- Nice visualization of cache records and how they link to each other
- Listing and pruning of cloud cache records

### Why is this important to you?

Users could use this to inspect their local cache - it would also be useful to have something like this during development of the engine.

**Comment by marcosnils** [comment]
thx for submitting this Justin. Another thing that I'd love to see is the GC policy of the engine as well? There's been several users that have been bitten by this and there's very little visibility about it.

**Comment by jedevc** [comment]
FYI @sipsma since you worked on https://github.com/dagger/dagger/pull/7767.

**Comment by helderco** [comment]
I think this is one of those cases where the cli command just calls the API for the implementation.

**Comment by sipsma** [comment]
Yeah some of the groundwork here has been released, specifically some preliminary raw API support for querying/pruning, e.g.
* Show all the cache entry metadata: echo '{daggerEngine{localCache{entrySet{entries{description diskSpaceBytes}}}}}' | dagger query
* Show just high level summaries of usage: echo '{daggerEngine{localCache{entrySet{entryCount diskSpaceBytes}}}}' | dagger query

https://github.com/dagger/dagger/pull/7646 is adding some more fine-grain pruning support so you can do things like "just prune cache mounts" or "just prune these exact layers", etc.

I'm saving the CLI frontend for more followups once the API is more fleshed out and confirmed stable, but will be a part of the overall https://github.com/dagger/dagger/issues/8004 effort
